pkg/provisioner/ironic/ironic.go

```diff
@@ -1613,6 +1617,33 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+	if ironicNode.PowerState == powerOn {
+		_, err := p.hardPowerOff()
+		if err != nil {
+			result.Dirty = true
+			switch err.(type) {
+			case HostLockedError:
+				p.log.Info("could not power off host, busy")
+				return retryAfterDelay(powerRequeueDelay)
+			default:
+				// For other errors continue with the delete
+				if p.host.Status.ErrorCount > maxErrorRetryCount {
+					result.Dirty = false
+					p.host.Status.ErrorCount = 0
+					p.host.Status.ErrorMessage = ""
```

Comment:
Is not anymore allowed to modify an host from within the provisioner (a copy is passed in the Provisioner factory). If the expected behavior in this case it's to give an `operationComplete()` should be returned and a proper log message traced

Reply:
Thanks for pointing that out. I have updated this code to use the new methods defined in result.go.